modules/services/getmail.nix

```diff
@@ -0,0 +1,205 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.programs.getmail;
+
+  retrieverModule = types.submodule ({config,...}: {
+    options = {
+      type = mkOption {
+        type = types.enum [
+          "SimplePOP3Retriever"
+          "SimplePOP3SSLRetriever"
+          "SimpleIMAPRetriever"
+          "SimpleIMAPSSLRetriever"
+        ];
+        default = "SimpleIMAPSSLRetriever";
+        description = "Type of the retriever.";
+      };
+
+      server = mkOption {
+        type = types.string;
+        default = "";
+        description = "The remote server.";
+      };
+
+      username = mkOption {
+        type = types.string;
+        default = "";
+        description = "The server username.";
+      };
+
+      password = mkOption {
+        type = types.nullOr types.string;
```

Comment:
Is null a valid value? Because down at the config file generation, first priority is `passwordCommand`, which might be null. When it's not null, `password` gets used instead, but if that is null too, it just uses the empty string as a password, which probably won't work. So I suggest making this of `types.str` without a default value and remove the `optionalString` at the config file generation -> when `passwordCommand` is null, `password` has to be set.